src/project/FileTreeView.js

```diff
@@ -569,6 +571,10 @@ define(function (require, exports, module) {
                 if (this.state.clickTimer !== null) {
                     this.clearTimer();
                 }
+                if("xd" === FileUtils.getFileExtension(this.myPath()).toLowerCase() && PreferencesManager.getViewState("OpenXDFileInXDApp")) {
+                    brackets.app.OpenDocumentWithXDApp(this.myPath(), function () {});
```

Comment:
See my comment on brackets-shell, you can make this callback optional by using _dummyCallback